src/test/java/com/es/phoneshop/service/implementation/ProductServiceImplTest.java

```diff
@@ -29,26 +32,21 @@ public class ProductServiceImplTest {
     @Before
     public void setup() {
         usd = Currency.getInstance("USD");
-        productService = new ProductServiceImpl();
-        productDao = ProductDaoImpl.getInstance();
+        productService = new ProductServiceImpl(productDao);
     }
 
     @Test
     public void getProduct_ValidId_ShouldReturnProductWithIdFromDao() {
         // given
-        final UUID id = productService.findProducts().get(0).getId();
-        final BigDecimal expectedPrice = new BigDecimal(100);
-        final String expectedDescription = "Samsung Galaxy S";
-        final int expectedStock = 100;
+        final Product product = new Product(UUID.randomUUID(), "sgs2", "Samsung Galaxy S II", new BigDecimal(200), usd, 0, "https://..");
+        final UUID id = product.getId();
 
         // when
-        Product product = productService.getProduct(id);
+        when(productDao.getProduct(id)).thenReturn(Optional.of(product));
+        productService.getProduct(id);
 
         // then
-        assertNotNull(product);
-        assertEquals(expectedPrice, product.getPrice());
-        assertEquals(expectedDescription, product.getDescription());
-        assertEquals(expectedStock, product.getStock());
+        verify(productDao, times(1)).getProduct(id);
     }
 
     @Test(expected = ProductNotFoundException.class)
```

Comment:
Don't see defined behavior for productDao#getProduct method via when statement and verification that productDao#getProduct method is invoked